storage/storage_test.go

```diff
@@ -264,20 +264,24 @@ func makeLayer(t *testing.T, compression archive.Compression) (ddigest.Digest, i
 			Typeflag:   tar.TypeReg,
 		})
 		if err != nil {
-			t.Fatalf("Error writing tar header: %v", err)
+			errs <- fmt.Errorf("Error writing tar header: %v", err)
```

Comment:
This is not sufficient, the previous code (intended to) terminate the goroutine. Now it will just send something to `errs` and continue.


Really, rather than repeat the error handling all over the place (and risk forgetting), I’d much rather split into _two_ separate (named, top-level) functions:
- One to be the main function of the goroutine, dealing _only_ with the error reporting
- One to actually do the work, and just return an ordinary `error` return value, not caring about goroutines at all. 

Compare `copy.diffIDComputationGoroutine` and `copy.computeDiffID`.